srai/neighbourhoods/_base.py

```diff
@@ -20,9 +20,20 @@ class Neighbourhood(ABC, Generic[IndexType]):
 
     The subclasses only need to implement the `get_neighbours` method, but can also override the
     `get_neighbours_up_to_distance` and `get_neighbours_at_distance` methods for performance
-    reasons. See the `H3Neighbourhood` class for an example.
+    reasons.  See the `H3Neighbourhood` class for an example. The class also provides a
+    `_handle_center` method, which can be used to handle the region itself.
     """
 
+    def __init__(self, include_center: bool = False) -> None:
```

Comment:
From the logic, I understand that returning `at_distance` always removes the center region. I'd move this parameter to methods `get_neighbours` and `get_neighbours_up_to_distance` with default value `False`. That would simplify the logic of `_handle_center` in my opinion.

If I had a use case where I need sometimes neighbours with myself and sometimes without, then I need to have two instances of this class or remove it explicitly.

Reply:
Not exactly - when you call `get_neighbours_up_to_distance`  and `include_center == True` then you will get the center region. If it's `False` then you get an empty set.

I can't really think of any example use case for sometimes having the center and sometimes discarding it but it still may not be a bad idea to have it as a parameter of the methods.
It doesn't complicate the API too much for the regular use case, so that still works, but having 2 instances for the other use case really seems wrong.